runner.go

```diff
@@ -104,9 +106,12 @@ func Run(options ...RunOption) {
 
 	tasks := runtime.registry.Tasks()
 
-	if listTasks {
+	if listTasks || listAllTasks {
 		outputString := "Run specified tasks with `taskrunner taskname1 taskname2`\nTasks available:"
 		for _, task := range tasks {
+			if listTasks && task.Hidden {
```

Comment:
This logic seems to assume that the options `listTasks` and `listAllTasks` are mutually exclusive.
I.e. they can both be false, or, only 1 can be true, but not both.

Is this worth enforcing when we parse the flags (return an error if both are true) so that this assumption will always be guaranteed?

Another way to do it might be to have a single variable `listMode` (instead of `listAllTasks` and `listTasks`) that is an enum:
```
ListModeNone
ListModeDefault
ListModeAll
```

That way we don't have to worry about how the 2 variables should interact with each other.

Reply:
Agreed that two boolean flags makes for some ambiguous behavior with `--list --listall`. However, we could just display an error saying that the two are mutually exclusive. 

One possible alternative would be to convert `--list` to a string which defaults to `default`. e.g. `--list=default` or `--list=all`. We would then need to initialize the variable to `"none"` instead of `""`.

Reply:
I made it fail if `--list` and `--listAll` are set. I prefer the route of multiple flags for now since there are only two variations of the list and this is the simplest addition